.github/workflows/on-demand-build-docker-image-deploy-preview.yml

```
@@ -72,6 +72,27 @@ jobs:
           cache: "yarn"
           cache-dependency-path: "app/client/yarn.lock"
 
+      - name: Install dependencies
+        if: steps.run_result.outputs.run_result != 'success'
+        run: yarn install --frozen-lockfile
+
+      - name: Set the build environment based on the branch
+        if: steps.run_result.outputs.run_result != 'success'
+        id: vars
+        run: |
+          echo "REACT_APP_ENVIRONMENT=DEVELOPMENT" >> $GITHUB_OUTPUT
+          if [[ "${{github.ref}}" == "refs/heads/master" ]]; then
+              echo "REACT_APP_ENVIRONMENT=PRODUCTION" >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{github.ref}}" == "refs/heads/release" ]]; then
+              echo "REACT_APP_ENVIRONMENT=STAGING" >> $GITHUB_OUTPUT
+          fi
```

Wouldn't this repeat the variable value? Is that okay? Can we make it put that variable only once? Like

```suggestion
          if [[ "${{github.ref}}" == "refs/heads/master" ]]; then
              echo "REACT_APP_ENVIRONMENT=PRODUCTION" >> $GITHUB_OUTPUT
          elif [[ "${{github.ref}}" == "refs/heads/release" ]]; then
              echo "REACT_APP_ENVIRONMENT=STAGING" >> $GITHUB_OUTPUT
          else
              echo "REACT_APP_ENVIRONMENT=DEVELOPMENT" >> $GITHUB_OUTPUT
          fi
```